File: CS Workshop/Programming/#2 Temperature Converter/temp.cpp
#include <string>
#include <iostream>

using namespace std;

int main(int argc, char** argv)
{
	// declare a float variable to store the fahrenheit temperature
	float fahrenheit;

	// declare a float variable to store the celsius temperature
	float celsius;
	
	// print a prompt to standard output (the screen)
	cout << "Enter temperature (in Fahrenheit): ";
	
	// read the next float the user enters from standard in (the keyboard).
	//   Notice this is different than the last program.  We used getline
	//   before to read a whole line, but now we're only interested in the
	//   next float variable.  What the keyboard enters is text (a string),
	//   but cin handles the conversion to a number type (like float) for us.
	cin >> fahrenheit;
	
	// calculate the celsius temperature
	celsius = (fahrenheit - 32) * 5/9;

	// print "Temperature is <temp> Celsius."
	cout << "Temperature is " << celsius << " Celsius.";
	
	// print a newline (like return) for good measure
	cout << endl;
	
	// return zero (means no error occurred)
	return 0;
}

File: CS Workshop/Programming/#2 Temperature Converter/temp.py
def main():
	
	# print the prompt and read the temperature variable from standard input (the keyboard)
	fahrenheit = raw_input("Enter temperature (in Fahrenheit): ")

	# convert the text from the keyboard into a number (float)
	fahrenheit = float(fahrenheit)

	# calculate the celsius temperature
	celsius = (fahrenheit - 32) * 5/9

	# print "Temperature is <temp> Celsius."
	#   in Python we need to convert celsius (currently a number) to a string using str()
	print "Temperature is " + str(celsius) + " Celsius."
	
	# return zero (means no error occurred)
	return 0
	

# if this script is run directly, run the main function above
if __name__ == "__main__":
    main()

